packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts

```diff
@@ -59,27 +59,27 @@ export const useDialogSurface_unstable = (
     }
   });
 
+  const backdropSlot = slot(backdrop, {
+    componentType: 'div',
+    required: open && modalType !== 'non-modal',
+    defaultProps: { 'aria-hidden': 'true' },
+    overrides: { onClick: handledBackdropClick },
+  });
+
   return {
-    components: {
-      backdrop: 'div',
-      root: 'div',
-    },
-    backdrop: resolveShorthand(backdrop, {
-      required: open && modalType !== 'non-modal',
-      defaultProps: {
-        'aria-hidden': 'true',
-        onClick: handledBackdropClick,
-      },
-    }),
-    root: getNativeElementProps(as ?? 'div', {
-      tabIndex: -1, // https://github.com/microsoft/fluentui/issues/25150
-      'aria-modal': modalType !== 'non-modal',
-      role: modalType === 'alert' ? 'alertdialog' : 'dialog',
-      'aria-labelledby': props['aria-label'] ? undefined : dialogTitleID,
-      ...props,
-      ...modalAttributes,
-      onKeyDown: handleKeyDown,
-      ref: useMergedRefs(ref, dialogRef),
-    }),
+    backdrop: backdropSlot,
+    root: slot<DialogSurfaceProps>(
```

Comment:
As `componentType` is a required param, is there a reason to keep it `options`? Did you consider following?

```ts
// Option 1
// slot(COMPONENT_TYPE, PROPS, OPTIONS)
slot('div', props.slot, { required: true })

// Option 2
// slot(OPTIONS)
slot({ elementType: 'div', props: props.slot, required: true })

```

Reply:
Option 1 I'm not a big fan, 3 arguments on a function brings a lot of options.

I'd be ok with Option 2, looks even simpler for me, one less argument

Reply:
here's one thing though.... on Option 2 you create the property `props`, that indicates we're going to be passing the props for the slot there, but we don't know if we have props, we only have `props.slot` which is a shorthand. soooo, should we call it `shorthand`?

Reply:
> should we call it shorthand?

Or may be just `value`...